src/tracks/ui/TextEditHelper.h

```diff
@@ -0,0 +1,83 @@
+/*!********************************************************************
+*
+ Audacity: A Digital Audio Editor
+
+ TextEditHelper.h
+
+ Vitaly Sverchinsky
+
+ **********************************************************************/
+
+#pragma once
+
+#include <wx/string.h>
+#include <wx/font.h>
+
+#include "UIHandle.h"
+#include <wx/colour.h>
+
+class wxDC;
+class wxMouseEvent;
+
+class AudacityProject;
+class TextEditDelegate;
+class Track;
+
+//Used as a helper object for drawing, editing
+//and text navigation in TrackPanel
+class TextEditHelper
+{
+    wxString mText;
+    wxFont mFont;
+    wxRect mBBox;
+
+    wxColor mTextSelectionColor;
+    wxColor mTextColor;
+
+    bool mRightDragging{ false };
+    int mOffset{ 0 };
+    int mInitialCursorPos{ 0 };
+    int mCurrentCursorPos{ 0 };
```

Comment:
Can you state in comments any intended invariant relations among these values?

Is it meant to be invariant that `mInitialCursorPos <= mCurrentCursorPos` ?

Reply:
Not neccessary that `mInitialCursorPos <= mCurrentCursorPos` it depends on direction in which selection is moving: from left to right or right to left. Variables have the same meaning as they do in LabelTrackView